server/odc-core/src/main/java/com/oceanbase/odc/core/datasource/SingleConnectionDataSource.java

```diff
@@ -136,13 +169,10 @@ public void addInitializer(@NonNull ConnectionInitializer connectionInitializer)
 
     @Override
     public void close() {
-        if (!Objects.isNull(this.connection)) {
-            try {
-                this.connection.close();
-            } catch (Throwable throwable) {
-                log.error("Failed to close the connection", throwable);
-            }
+        if (Objects.nonNull(keepAliveScheduler) && !keepAliveScheduler.isTerminated()) {
+            shutdownKeepAliveScheduler();
```

Comment:
we may try-catch this to avoid possible exception throwing

Reply:
done